cmake/external/leveldb_patch.py

```diff
@@ -50,6 +53,10 @@ def main() -> None:
   with cmakelists_txt_file.open("wt", encoding="utf8") as f:
     f.writelines(patched_lines)
 
+  if additional_patch_file:
```

Comment:
@jonsimantov I just found a pretty annoying bug with this code: it needs to be re-entrant. That is, if you re-run "cmake" to update the build files this will attempt to re-apply the patch, then fail. The only recovery is to delete your cmake "build" directory and regenerate.

Typically these "re-entrant" semantics are achieved by creating a "stamp" file. After successfully applying the patch you'd create a "stamp" file to mark the patch as having been applied. Then, before applying the patch, you check the "stamp" file and only apply the patch if it has not previously been applied.

Here is an example from the python virtualenv setup code that uses a "stamp" file to avoid re-creating the virtualenv if it's already created: https://github.com/firebase/firebase-ios-sdk/blob/52ebbbf19d23baeb2b8c08b4b3d2b7195d5074cd/cmake/python_setup.cmake#L96-L117